vimeo-networking/src/main/java/com/vimeo/networking/model/Video.java

```diff
@@ -796,21 +796,44 @@ public boolean isSvod() {
     }
 
     /**
-     * Helper to determine if the video is part of the SVOD library and that the user currently has access to it.
-     * This will be true if the user has purchased an SVOD subscription that is active.
+     * Helper to determine if this video is theoretically playable. Note that there may be device limitations
+     * that cause a video to fail to play back in practice.
      *
-     * @return true if this video can be accessed with an SVOD subscription and the user's subscription is active
+     * @return true if this video can be played, false otherwise
      */
-    public boolean canAccessVideoInSvod() {
-        return getSvodPurchaseDate() != null;
+    public boolean isPlayable() {
+        return getPlayStatus() == Play.Status.PLAYABLE;
     }
 
+    /**
+     * Helper to determine if this is a trailer that is associated with an SVOD title
+     *
+     * @return true if this is a trailer that is associated with an SVOD title, false otherwise
+     */
+    public boolean isSvodRelatedTrailer() {
+        return isTrailer() && getSvodInteraction() != null;
+    }
+
+
+    /**
+     * Helper to determine if this is an SVOD video (not a trailer) that has not been purchased.
+     * @return true if it is an unpurchased video, false otherwise
+     */
+    public boolean isUnpurchasedSvod() {
+        return !isTrailer() &&
+               isSvod() &&
+               !isPlayable() &&
+               getSvodInteraction() != null &&
+               getSvodInteraction().getPurchaseDate() == null;
```

Comment:
Could `getSvodInteraction() != null && getSvodInteraction().getPurchaseDate() == null` be removed in favor of `getSvodPurchaseDate() == null` since we also confirm that it is an Svod?